### PR TITLE
cdb2api Use our local timezone with localtime.

### PR DESCRIPTION
**Pull request by jmleddy**

### CODE REVIEW

File: cdb2api/cdb2api.c [hunk 1]
```diff
@@ -1965,9 +1967,15 @@ static int cdb2_send_query(cdb2_hndl_tp *hndl, SBUF2 *sb, char *dbname,
 
     if (env_tz == NULL) {
         env_tz = DB_TZNAME_DEFAULT;
-    }
+    } else if (strcmp(env_tz, "localtime") == 0) {
```
Comment (mponomar) [inline cdb2api/cdb2api.c]: Make those calls here, once?

https://github.com/bloomberg/comdb2/blob/master/cdb2api/cdb2api.c#L95

I'm not a huge fan of calls that set state outside of comdb2's domain, but I guess this is the portable way to do it...

Comment (jmleddy) [inline cdb2api/cdb2api.c]: Yeah that makes more sense.

File: cdb2api/cdb2api.c [hunk 2]
```diff
@@ -1965,9 +1967,15 @@ static int cdb2_send_query(cdb2_hndl_tp *hndl, SBUF2 *sb, char *dbname,
...
+        tzset();
+        int tzlen = strlen(tzname[0]) + strlen(tzname[1]) + 4;
+        int offset = timezone / (60 * 60);
 
-    sqlquery.tzname = env_tz;
+        local_tz = malloc(tzlen);
```
Comment (riverszhang89) [inline cdb2api/cdb2api.c]: `alloca` seems better than `malloc` here.

**Comment by jmleddy** [comment]
The comdb2 server does not understand POSIX timezones. Before this can work, we would have to support POSIX timezones in the server. Alternately, we can cut the /usr/share/zoneinfo part of the localtime link.

**Pull request closed without merging**